svc/GridAutosizeService.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2020 Extremely Heavy Industries Inc.
+ */
+
+import {HoistService} from '@xh/hoist/core';
+import {throwIf, stripTags} from '@xh/hoist/utils/js';
+import {isFunction, isNil, isFinite, sortBy, reduce} from 'lodash';
+
+@HoistService
+export class GridAutosizeService {
+
+    _canvasContext;
+    _cellEl;
+
+    autoSizeColumns({gridModel, colIds = []}) {
```

Comment:
Need docstring on method and/or on this class.   I think it does make sense to make this a service -- its quite similar to GridExportService, and you could also imagine applications replacing it or enhancing it in some way,